tests/unit/cluster/multidb.tcl

```diff
@@ -0,0 +1,401 @@
+        }
+        
+    }
+
+    test "Validate slot statistics using cluster countkeysinslot and cluster getkeysinslot" {        
+        set primary_id 0
+        set keys_per_db 100
+
+        # Set keys in all DBs with same hash slot
+        for {set db 0} {$db < 16} {incr db} {
+            R $primary_id select $db
+            for {set i 0} {$i < $keys_per_db} {incr i} {
+                set key "{x}key$i"
+                set val "hello_db${db}_{x}key$i"
+                R $primary_id set $key $val
+                assert_equal [R $primary_id get $key] $val
+            }
+        }
+
+        # Get a random key to determine the slot, which will be identical to all keys
+        set random_key [R $primary_id randomkey]
+        assert_not_equal $random_key ""
+        
+        set slot [R $primary_id cluster keyslot $random_key]
+        assert_not_equal $slot ""
+
+        # Validate slot key distribution in each database
+        for {set db 0} {$db < 16} {incr db} {
+            R $primary_id select $db
+            set keys_in_slot [R $primary_id cluster countkeysinslot $slot]
+            # Since all keys are mapped to a single slot, the number of keys in the currently selected db should match $keys_per_db
+            assert_equal $keys_in_slot $keys_per_db 
+        }
+
+        # Make sure slot-stats counts keys in all databases
+        set slotstats [R $primary_id CLUSTER SLOT-STATS  SLOTSRANGE  $slot $slot]    
+        regexp {^\{(\d+)\s+\{key-count\s+(\d+)} $slotstats _ parsed_slot parsed_count
+        set total_items [expr $keys_per_db * 16]
+        assert_equal $parsed_slot $slot
+        assert_equal $parsed_count $total_items
+
+        # Verify key retrieval by slot for each database
+        for {set db 0} {$db < 16} {incr db} {
+            R $primary_id select $db
+            set slot_keys [R $primary_id cluster getkeysinslot $slot $keys_per_db]
+            assert_equal [llength $slot_keys] $keys_per_db
+            foreach key $slot_keys {
+                if {![regexp {^\{x\}key\d+$} $key]} {
+                    error "Key format mismatch: $key"
+                }
+                set expected_val "hello_db${db}_$key"            
+                assert_equal [R $primary_id get $key] $expected_val
+            }
+        }
+
+        # Delete all keys and verify slot emptiness
+        for {set db 0} {$db < 16} {incr db} {
+            R $primary_id select $db
+            for {set i 0} {$i < $keys_per_db} {incr i} {
+                set key "{x}key$i"
+                R $primary_id del $key
+            }
+            assert_equal [R $primary_id dbsize] 0
+        }
+
+        # Ensure the slot is empty
+        set remaining_keys [R $primary_id cluster countkeysinslot $slot]
+        assert_equal $remaining_keys 0
+    }
+
+
+
+    test "Replication: Write to multiple databases and verify replica" {
+        set primary_id 0    
+
+        set replica [get_my_replica [R $primary_id cluster nodes]]    
```

Comment:
```suggestion
        set replica [get_my_replica [R $primary_id cluster replicas]]    
```
Minor thing, but there is a function that just returns the replicas of a given node.